### PR TITLE
Created a basic UI to outline what we have planned so far

### PR DESCRIPTION
**Pull request by WoppleSupreme**
I'm not sure if the class extensions that make the buttons rounded are in the right place, but I wasn't sure where else to put them. Other than that, all we need to do is to migrate John's 'RECALL' button functionality to this one, and make it all nice and pretty.

We still need a view controller for the hamburger menu, and to actually add the menu button.

Screen Shot 2019-08-09 at 4 15 58 PM

### CODE REVIEW

**Comment by WoppleSupreme** [comment]
I added the hamburger button to the navigation bar, and made an outlet for it. Then, realized it should be an action and not an outlet. 

Screen Shot 2019-08-09 at 4 21 44 PM

File: Recipick/View/RecipeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//  RecipeViewController.swift
+//  Recipick
+//
+//  Created by Brian Winn on 8/9/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import UIKit
+
+class RecipeViewController: ViewController {
```
Comment (prrenfrew) [inline Recipick/View/RecipeViewController.swift]: Final class

Comment (prrenfrew) [reply Recipick/View/RecipeViewController.swift]: Why are you subclassing ViewController as opposed to UIViewController?

File: Recipick/View/RecipeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  @IBOutlet weak var vegetarianLabel: UILabel!
+  @IBOutlet weak var veganLabel: UILabel!
+  @IBOutlet weak var recipeTitle: UILabel!
+  
+  @IBOutlet weak var recipeInstructions: UITableView!
+  
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+  }
```
Comment (prrenfrew) [inline Recipick/View/RecipeViewController.swift]: Don't override functions if you are not adding any functionality.

File: Recipick/View/RoundedButton.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class RoundedButton: UIButton {
+  
+  @IBInspectable var cornerSize: CGFloat = 30 {
+    didSet{
+      setUpView()
+    }
+  }
+  
+  override func prepareForInterfaceBuilder() {
+    setUpView()
```
Comment (prrenfrew) [inline Recipick/View/RoundedButton.swift]: Why is super not called?

File: Recipick/View/RoundedButton.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  
+  func setUpView() {
```
Comment (prrenfrew) [inline Recipick/View/RoundedButton.swift]: Is there a reason this is not private?

File: Recipick/View/RoundedImage.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//  Recipick
+//
+//  Created by Brian Winn on 8/9/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import UIKit
+@IBDesignable
+
+class RoundedImage: UIImageView {
```
Comment (prrenfrew) [inline Recipick/View/RoundedImage.swift]: Final class

File: Recipick/View/RoundedImage.swift [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  
+  @IBInspectable var cornerSize: CGFloat = 25 {
+    didSet{
+      setUpView()
+    }
+  }
+  
+  override func prepareForInterfaceBuilder() {
+    setUpView()
```
Comment (prrenfrew) [inline Recipick/View/RoundedImage.swift]: Why aren't you calling super?

File: Recipick/View/RecipeViewController.swift (hunk 2 above)
Comment (WoppleSupreme) [reply Recipick/View/RecipeViewController.swift]: I just had these in there out of habit. I'll go in and delete them.

File: Recipick/View/RecipeViewController.swift (hunk 1 above)
Comment (WoppleSupreme) [reply Recipick/View/RecipeViewController.swift]: That was an oopsie. It's fixed now.

File: Recipick/View/SearchRandomViewController.swift [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+class SearchRandomViewController: ViewController {
+  @IBAction func hamburgerMenuBarButton(_ sender: Any) {
+  }
+  
+  @IBOutlet weak var recipeCollectionView: UICollectionView!
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+  }
```
Comment (prrenfrew) [inline Recipick/View/SearchRandomViewController.swift]: Remove overridden functions if you are just calling super.

File: Recipick/View/RoundedButton.swift (hunk 3 above)
Comment (WoppleSupreme) [reply Recipick/View/RoundedButton.swift]: I had this from a tutorial that I was using from the SpeechRecognition Project that I tried, emphasis on tried, to do. This was just the way it was done, but I can try to do super.setUpView()

File: Recipick/View/RoundedButton.swift (hunk 4 above)
Comment (WoppleSupreme) [reply Recipick/View/RoundedButton.swift]: That was a mistake on my part. I'll go in and fix it.

File: Recipick/View/RoundedImage.swift (hunk 5 above)
Comment (WoppleSupreme) [reply Recipick/View/RoundedImage.swift]: That's fixed now.

File: Recipick/View/RoundedImage.swift (hunk 6 above)
Comment (WoppleSupreme) [reply Recipick/View/RoundedImage.swift]: UIButtons don't have a setUpView() method, so I can't call super. I don't know why it's an override, though, if the superclass doesn't have one.

Comment (WoppleSupreme) [reply Recipick/View/RoundedImage.swift]: setUpView() is a IBDesignable function, not a UIButton function. I guess that it just knows where to override?

File: Recipick/View/RecipeViewController.swift [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+final class RecipeViewController: UIViewController {
+  @IBOutlet weak var recipeImageView: RoundedImage!
+  @IBOutlet weak var dairyLabel: UILabel!
+  @IBOutlet weak var nutLabel: UILabel!
+  @IBOutlet weak var glutenLabel: UILabel!
+  @IBOutlet weak var vegetarianLabel: UILabel!
+  @IBOutlet weak var veganLabel: UILabel!
+  @IBOutlet weak var recipeTitle: UILabel!
+
```
Comment (prrenfrew) [inline Recipick/View/RecipeViewController.swift]: Why this blank line?

Comment (WoppleSupreme) [reply Recipick/View/RecipeViewController.swift]: To have something for me to do for today during class.

File: Recipick/View/RoundedButton.swift (hunk 3 above)
Comment (prrenfrew) [reply Recipick/View/RoundedButton.swift]: Just google whether you should. Come back after you have an answer.

Comment (WoppleSupreme) [reply Recipick/View/RoundedButton.swift]: I DO need to call super.prepareForInterfaceBuilder(), and I am now doing so before the setUpView function.

File: Recipick/View/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+//
+//  Created by MAC on 8/7/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import UIKit
+
+final class ViewController: UIViewController {
+  @IBOutlet weak var searchBar: UISearchBar!
+  @IBOutlet weak var recipeTable: UITableView!
```
Comment (prrenfrew) [inline Recipick/View/ViewController.swift]: Names of properties should reflect what it is. For this case, recipeTable is actually an table view.

**Comment by prrenfrew** [comment]
For your empty functions and other places that require more things to be done, please leave TODO comments there.

**Pull request closed without merging**